altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -250,43 +524,83 @@ impl CR2 {
 /// CR3
 // ------------------------------------
 
-#[derive(Copy, Clone)]
+/// Defines the possible HardwareFlowControl configurations for the Usart.
+/*  None: No hardware flow control
+ *  Rts: Request to Send enabled
+ *  Cts: Clear to Send enabled
+ *  All: Both are enabled.
+ */
+#[derive(Copy, Clone, Debug)]
 pub enum HardwareFlowControl {
     None,
-    /// Request to Send
     Rts,
-    /// Clear to Send
     Cts,
-    /// Both
-    RtsCts,
+    All,
 }
 
-#[derive(Copy, Clone)]
+/// Defines the base address variable for control register three.
+#[derive(Copy, Clone, Debug)]
 struct CR3 {
     base_addr: *const u32,
 }
 
+/// Implements the Register trait for Usartx_CR3.
+/// Stores base address for control register three, which is the address
+/// of the Usart being used to access this register. Uses the base address
+/// combined with the register offset to calculate the register address.
```

Comment:
I'm running out of things to say, but I want to put a comment here just so we don't forget about any of these